OpenSim/Common/Component.h

```diff
@@ -198,19 +198,16 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     Component(const std::string& aFileName,
         bool aUpdateFromXMLNode = true) SWIG_DECLARE_EXCEPTION;
 
+    /** Construct Component as a subcomponent member of another Component.
+        Automatically marks the component as a subcomponent of the owner.
+        A subcomponent MUST be named. **/
+    explicit Component(const std::string& name, Component* owner=nullptr);
```

Comment:
> Since the in-class member initializer is called during construction of the parent (owner), is it permissible to call a method on the parent as a side effect of the in-class member initialization?

Oh. Cheeky. Probably not. In that case the address of the parent is probably the only thing you can count on. But then I think this constructor needs a new comment explaining what it is for and how to use it. And that second argument should definitely *not* be optional!